ts-node-plugin/index.js

```diff
@@ -0,0 +1,38 @@
+const {transform} = require("../dist");
+
+const ModuleKindCommonJS = 1;
+
+/**
+ * ts-node transpiler plugin
+ */
+function create(createOptions) {
+  const {nodeModuleEmitKind} = createOptions;
+  const {module} = createOptions.service.config.options;
+  return {
+    transpile(input, transpileOptions) {
+      const {fileName} = transpileOptions;
+      const transforms = ["typescript"];
+      if (module === ModuleKindCommonJS || nodeModuleEmitKind === "nodecjs") {
+        transforms.push("imports");
+      }
+      if (fileName.endsWith(".tsx") || fileName.endsWith(".jsx")) {
+        transforms.push("jsx");
+      }
+
```

Comment:
Another thing to think about is what to do about the TS options `importsNotUsedAsValues: preserve` and `preserveValueImports: true`. Currently, Sucrase just implements the default behavior of eliding type-only (or unused) imports, so there may be a difference in execution behavior if Sucrase elides an import due to all names being type-only, but the import was actually important to run due to import side-effects or due to the order of import execution. To me, this feels like a rare enough situation that maybe it's fine to try to run the code anyway, but a stricter approach would be to give an error/warning if this configuration is detected. My current leaning is to ignore those configs for now. For the future, I'm certainly interested in a "don't elide anything" mode in Sucrase, and can maybe hook up the options then.

@cspotcode it looks to me like the swc integration also ignores those settings.